lib/Sema/CSSimplify.cpp

```diff
@@ -3554,18 +3554,9 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
   // Value member lookup has some hacks too.
   if (shouldAttemptFixes() && baseObjTy->getOptionalObjectType()) {
     // If the base type was an optional, look through it.
-    
-    // Determine whether or not we want to provide an optional chaining fixit or
-    // a force unwrap fixit.
-    bool optionalChain;
-    if (!getContextualType())
-      optionalChain = !(Options & ConstraintSystemFlags::PreferForceUnwrapToOptional);
-    else
-      optionalChain = !getContextualType()->getOptionalObjectType().isNull();
-    auto fixKind = optionalChain ? FixKind::OptionalChaining : FixKind::ForceOptional;
-
-    // Note the fix.
-    if (recordFix(fixKind, locator))
+
+    // We're unwrapping the base to perform a member access.
+    if (recordFix(Fix::getUnwrapOptionalBase(*this, member), locator))
```

Comment:
I’m confident the rest of this can be pulled out of the typing rule.  Removing it altogether affects like 4 diagnostics in the test suite.

The only reason it’s here is because it is, by definition, operating on chains.